src/packer/skyline_packer.rs

```diff
@@ -155,39 +151,39 @@ impl<B: Buffer2d> SkylinePacker<B> {
     }
 }
 
-impl<B: Buffer2d> Packer<B> for SkylinePacker<B> {
-    fn pack(&mut self, buf: &Buffer2d) -> Option<Rect> {
-        let (mut width, mut height) = buf.dimensions();
-        width += self.margin;
-        height += self.margin;
+impl<P: Pixel> Packer for SkylinePacker<P> {
+    type Pixel = P;
 
-        match self.find_skyline(width, height) {
-            Some((i, mut rect)) => {
-                if width == rect.w {
-                    self.buf.patch(rect.x, rect.y, buf);
-                } else {
-                    self.buf.patch_rotated(rect.x, rect.y, buf);
-                }
+    fn pack(&mut self, key: String, texture: &Texture<Pixel=P>) -> Option<Frame> {
```

Comment:
Because we need some ways to retrieve the frame after packing texture. Do you have suggestions?